### ISSUE TITLE
Changing default calendar and default addressbook location.

### ISSUE THREAD

**Issue by ericavijay** [Kozea/Radicale#320]
We have a number of users with calendar and addressbook in ics and vcf file format respectively.  All the files are stored in 
/srv/pool0/$USER/$USER.ics
/srv/pool0/$USER/$USER.vcf

These files are used by the current web front end by all users. So Ideally we map these in xmlutils.py - but if this can be configured in the config file it will make it easy for all users to do autodiscovery and end up in the same folder. Here is my hack on Raicale 0.9 which seems to work universally.  For some reason the MAc calendar client doing discovery of "current-user-principal" is not being recognized as a "addressbook" resource lookup. So I had to further hack the xmlutils to look for "addressbook" in user-agent for the discovery to work right. 
Here is my patch/hack 
Radicale-0.9
--- 304,314 ----
          elif tag == _tag("D", "current-user-principal") and user:
              tag = ET.Element(_tag("D", "href"))
              if item.resource_type == "addressbook":
!                 tag.text = _href("/%s/%s.vcf/" %(user,user))
              else:
!                 tag.text = _href("/%s/%s.ics/" %(user,user))
!             if headers.get('HTTP_USER_AGENT','Unknown User agent').lower().find("addressbook") > -1:
!                 tag.text = _href("/%s/%s.vcf/" %(user,user))

Do you have any solutions or suggestions with the newer version of Radicale I can work with?

Thanks

**Comment by liZe** [comment]
Did you try to play with the `well-known` section of the documentation?

**Comment by ericavijay** [comment]
I saw a contribution to well-known have not found the documentation - can you narrow it down as to where it is ?  some sample configurations?

**Comment by liZe** [comment]
Sorry, the documentation was not up to date, I've just updated it. You can find the configuration keys and the default values in the `well-known` section of the configuration file: http://radicale.org/user_documentation/#idmain-configuration-file

It should be enough for you, but some additional documentation would probably be useful.

**Comment by ericavijay** [comment]
Hello Guillaome,

I spent a lot of time playing with these configurations.  the "well-known" is a bit of trouble as it causes certain clients (non-Mac) to just crash. I also played with permissions to see if I can hide all other files but "%user.ics" for calendar requests and "%user.vcf" for Vcard requests. This also seems tricky at the moment

Ideally for my users it will be helpful it they just hit the calendar URL and it start mapping the response to start using
https://calendar.stime.net/%user/%user.ics
for CalDAV
https://calendar.stime.net/%user/%user.vcf 
for CardDAV

Basically hide ".vcf" file for calendar requests and ".ics" for vcard resource request.  I can code this and give you a patch but I want to make sure I am not missing something you already have in the  configuration setup. 

Thanks for your input and ideas!

**Comment by liZe** [comment]
@ericavijay The latest changes include removing well-known requests and cleaning various URL-related properties. Everything should be fine in 2.0, I've tested Apple's Calendar and it works very well :smile:. Feel free to reopen this ticket if you have the problem with the current master branch!
